pkg/auth/iam/iam.go

```diff
@@ -63,6 +63,9 @@ func NewFilter(client iam.Client) *Filter {
 
 // NewFilterWithOptions creates new Filter instance with Options
 func NewFilterWithOptions(client iam.Client, options *FilterInitializationOptions) *Filter {
+	if options == nil {
+		return &Filter{iamClient: client, options: &FilterInitializationOptions{}}
```

Comment:
actually I'm a little confused by the 2 new, if the developer add nil to this new method, why not use the above method?

Reply:
updated, reuse the above `NewFilter`